### PR TITLE
giflib: add libs property

### PR DESCRIPTION
**Pull request by glennpj**

### CODE REVIEW

File: var/spack/repos/builtin/packages/giflib/package.py [hunk 1]
```diff
@@ -44,6 +44,10 @@ def install_targets(self):
         targets = ['install'] + self.prefix_and_libversion_args()
         return targets
 
+    @property
+    def libs(self):
+        return find_libraries(['libgif'], root=self.prefix, recursive=True)
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/giflib/package.py]: Can we avoid recursion here? There are issues when recursion starts from the prefix and we use externals.

**Review by alalazo — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/giflib/package.py (hunk 1 above)
Comment (glennpj) [reply var/spack/repos/builtin/packages/giflib/package.py]: Hmm, is this sort of a catch-22? It is not known _a priori_ what the library directory is, `lib` or `lib64`, as that is part of what is being determined by `find_libraries`. What would I set the `root` to if not using recursion?

Comment (alalazo) [reply var/spack/repos/builtin/packages/giflib/package.py]: @glennpj Can we do something like:
```python
libs = find_libraries(['libgif'], root=self.prefix.lib) or find_libraries(['libgif'], root=self.prefix.lib64)
```
?

Comment (glennpj) [reply var/spack/repos/builtin/packages/giflib/package.py]: We could, and I thought of something like above. It probably covers most cases but it seems like there could be libraries in a directory named differently, which would be covered by recursion. I guess I am thinking more generally, not necessarily the giflib case, and perhaps over thinking. 

Thanks.

**Review by alalazo — APPROVED** [review]

**Pull request merged**